src/client.rs

```diff
@@ -11,28 +11,30 @@ use tokio::io::{AsyncReadExt, AsyncWrite, AsyncWriteExt};
 use tracing::debug;
 
 use crate::protocol::{read_lp_data, write_lp, Handshake, Request, Res, Response};
-use crate::tls::{self, Keypair};
+use crate::tls::{self, Keypair, PeerId};
 
 const MAX_DATA_SIZE: usize = 1024 * 1024 * 1024;
 
 #[derive(Clone, Debug)]
 pub struct Options {
     pub addr: SocketAddr,
+    pub peer_id: Option<PeerId>,
 }
 
 impl Default for Options {
     fn default() -> Self {
         Options {
             addr: "127.0.0.1:4433".parse().unwrap(),
+            peer_id: None,
         }
     }
 }
 
 /// Setup a QUIC connection to the provided server address
-async fn setup(server_addr: SocketAddr) -> Result<(Client, Connection)> {
+async fn setup(server_addr: SocketAddr, peer_id: Option<PeerId>) -> Result<(Client, Connection)> {
```

Comment:
```suggestion
async fn setup(opts: Options) -> Result<(Client, Connection)> {
```
?
We pass all the options here and never refer to them again in the `run` function.